tests/pytests/test_tdscf_excitations.py

```diff
@@ -26,1260 +25,219 @@
+    pytest.param( "H2O2", 'RHF-1',     'wB97X',  'TDA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_singlet, TDA]),
+    pytest.param( "H2O2", 'RHF-3',     'wB97X',  'RPA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_triplet, RPA]),
+    pytest.param( "H2O2", 'RHF-3',     'wB97X',  'TDA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_triplet, TDA]),
+    pytest.param( "METHYLOXIRANE", 'RHF-1',     'wB97X',  'RPA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_singlet, RPA]),
+    pytest.param( "METHYLOXIRANE", 'RHF-1',     'wB97X',  'TDA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_singlet, TDA]),
+    pytest.param( "METHYLOXIRANE", 'RHF-3',     'wB97X',  'RPA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_triplet, RPA]),
+    pytest.param( "METHYLOXIRANE", 'RHF-3',     'wB97X',  'TDA',  'cc-pvdz', marks=[hyb_gga_lrc, RHF_triplet, TDA]),
 ]) # yapf: disable
-def test_tdscf(ref, func, ptype, basis, expected, wfn_factory, solver_funcs, engines):
-    if (ref == 'RHF-1') or (func == "HF"):
-        # RHF-singlet everything works and TDHF/CIS works for RHF-triplet, UHF
-        pass
-    elif (ref == 'RHF-3'):
+def test_tdscf(mol, ref, func, ptype, basis, molecules, reference_data):
+    # expected failures
+    if (ref == 'RHF-3') and (func != "HF"):
         pytest.xfail("RKS Vx kernel only Spin Adapted for Singlet")
     elif (ref == 'UHF' and func != 'SVWN'):
-        pytest.xfail("UKS Vx kernel bug for non-lda")
-
-    ### setup
-    # Look up expected in fixture (easier to read)
-    exp_lookup = "{}-{}-{}-{}".format(ref, func, ptype, basis)
-    exp_low_per_sym = {'A1': 100, 'A2': 100, 'B1': 100, "B2": 100}
-    for x in expected[exp_lookup]:
-        exp_low_per_sym[x['sym']] = min(x['e'], exp_low_per_sym[x['sym']])
-
-    exp_energy_sorted = np.array([x['e'] for x in expected[exp_lookup]])
-    exp_energy_sorted.sort()
-
-    # get wfn (don't use symmetry b/c too slow)
-    wfn = wfn_factory(ref, func, basis, nosym=True)
-    # select solver function (TDA->davidson/RPA->hamiltonian)
-    solver = solver_funcs[ptype]
-
-    # build engine
-    engine = engines[ref](wfn, ptype)
-
-    # skipping the entrypoint, just call the solver
-    out = solver(
-        engine=engine,
-        guess=engine.generate_guess(16),
-        max_vecs_per_root=10,
-        nroot=4,
-        verbose=1,
-        maxiter=30,
-        e_tol=1.0e-7,
-        r_tol=1.0e-5,
-        schmidt_tol=1.0e-12)
-
-    test_vals = out[0]
-    stats = out[-1]
-    assert stats[-1]['done'], "Solver did not converge"
-
-    for i, my_v in enumerate(test_vals):
-        ref_v = exp_energy_sorted[i]
-        assert compare_values(ref_v, my_v, 4, "{}-{}-{}-ROOT-{}".format(ref, func, ptype, i + 1))
+        pytest.xfail("UKS Vx kernel bug for non-LDA")
+
+    molecule = molecules[mol]
+    psi4.set_options({'scf_type': 'pk', 'e_convergence': 8, 'd_convergence': 8, 'save_jk': True})
+    if ref == "UHF":
+        psi4.set_options({'reference': 'UHF'})
+    molecule.reset_point_group('c1')
```

Comment:
goal is to remove this line when backtransformation issue resolved?

Reply:
The tests are all run with PK, so that can be removed. AJ had left a comment saying that running with symmetry was "slow" and I just kept the tests as they were.

Reply:
Just tried removing that and there's two problems:
1. The reference values were obtained without symmetry, so the ordering of states will differ. I can regenerate this data easily in a few days. However, this requires seeking 8 states at least (to make sure we cover all irreps reasonably well) and the tests become accordingly longer.
2. UHF+PK+Symmetry lead to convergence failures too. I thought I had tested this but apparently not...

Reply:
Issue no. 2 is actually a too tight `r_convergence`: the solver stalls for the first irrep already. Not sure how to solve it...